lib/rucio/common/types.py

```diff
@@ -99,3 +99,43 @@ class InternalScope(InternalType):
     '''
     def __init__(self, scope, vo='def', fromExternal=True):
         super(InternalScope, self).__init__(value=scope, vo=vo, fromExternal=fromExternal)
+
+
+class RSEDomainLANDict(TypedDict):
+    read: int
+    write: int
+    delete: int
+
+
+class RSEDomainWANDict(TypedDict):
+    read: int
+    write: int
+    delete: int
+    third_party_copy_read: int
+    third_party_copy_write: int
```

Comment:
All protocol priorities should be `Optional[int]` because `None` is a possible value (it’s not yet reflected in the models).